holoviews/plotting/bokeh/callbacks.py

```diff
@@ -547,14 +567,176 @@ def _process_msg(self, msg):
+        elif Panel is None:
+            raise VersionError("Popup requires Bokeh >= 3.4")
+
+        close_button = Button(label='', stylesheets=[r"""
+        :host(.bk-Button) {
+            width: 100%;
+            height: 100%;
+            top: -1em;
+        }
+        .bk-btn, .bk-btn:hover, .bk-btn:active, .bk-btn:focus {
+            background: none;
+            border: none;
+            color: inherit;
+            cursor: pointer;
+            padding: 0;
+            outline: none;
+            box-shadow: none;
+            position: absolute;
+            top: 0;
+            right: 0;
+        }
+        .bk-btn::after {
+            content: '\2715';
+        }
+        """])
+        self._panel = Panel(
+            position=XY(x=np.nan, y=np.nan),
+            anchor="top_left",
+            elements=[close_button],
+            visible=False
+        )
+        close_button.js_on_click(CustomJS(args=dict(panel=self._panel), code="panel.visible = false"))
+
+        geom_type = self.geom_type
+        self.plot.state.on_event('selectiongeometry', self._populate)
+        self.plot.state.js_on_event('selectiongeometry', CustomJS(
+            args=dict(panel=self._panel),
+            code=f"""
+            export default ({{panel}}, cb_obj, _) => {{
+              const el = panel.elements[1]
+              if ((el && !el.visible) || ({geom_type!r} !== 'any' && cb_obj.geometry.type !== {geom_type!r})) {{
+                 return
+              }}
+              let pos;
+              if (cb_obj.geometry.type === 'point') {{
+                pos = {{x: cb_obj.geometry.x, y: cb_obj.geometry.y}}
+              }} else if (cb_obj.geometry.type === 'rect') {{
+                pos = {{x: cb_obj.geometry.x1, y: cb_obj.geometry.y1}}
+              }} else if (cb_obj.geometry.type === 'poly') {{
+                pos = {{x: Math.max(cb_obj.geometry.x), y: Math.max(cb_obj.geometry.y)}}
+              }}
+              if (pos) {{
+                panel.position.setv(pos)
+              }}
+            }}""",
+        ))
+        self.plot.state.elements.append(self._panel)
+
+    def _get_position(self, event):
+        if self.geom_type not in ('any', event.geometry['type']):
+            return
+        elif event.geometry['type'] == 'point':
+            return dict(x=event.geometry['x'], y=event.geometry['y'])
+        elif event.geometry['type'] == 'rect':
+            return dict(x=event.geometry['x1'], y=event.geometry['y1'])
+        elif event.geometry['type'] == 'poly':
+            return dict(x=np.max(event.geometry['x']), y=np.max(event.geometry['y']))
+
+    def _schedule_populate(self, event):
+        # IMPORTANT: NEED FIX BY PHILIPP
+        # - doesn't work on server; tap x/y updates late; so results in None, None
+        # - if pan is active, and tap, it gets dragged
+        state.execute(partial(self._populate, event), schedule=True)
+
+    def _populate(self, event):
+        if not event.final:
```

Comment:
```suggestion
        if not event.final or event.geometry["type"] not in self.geom_type:
```

This is to fix the problem I observed in https://github.com/holoviz/holoviews/pull/6168#issuecomment-2044671358 (Maybe need extra safeguards).

Reply:
Thanks, that works.